src/deps_bin/impl/bin.clj

```diff
@@ -59,15 +89,11 @@
     {:success false :reason :no-name}
 
     :else
-    (let [name (if windows? (str name ".bat") name)
-          bin-file (io/file name)]
-      (println "Creating standalone executable:" name)
-      (write-bin name jar (preamble options))
-      (when-not skip-realign
-        (println "Re-aligning zip offsets...")
-        (repair-zip-with-preamble-bytes bin-file))
-      {:success true
-       :bin-path (.getCanonicalPath bin-file)})))
+    (let [platforms     (coerce-platforms platforms)
+          emitted-paths (mapv #(emit-bin! % options) platforms)]
+      (if (= (count emitted-paths) 1)
+        {:success true :bin-path (first emitted-paths)}
+        {:success true :bin-paths emitted-paths}))))
```

Comment:
Factored out existing code into a single fn `emit-bin!` that both side effects and returns. I needed the return for the last bit, because I wanted to keep `:bin-path` when only a single bin has been emitted – keeping existing users of this `build-bin` happy. Otherwise, I'd have used `doseq` over `mapv`.

Reply:
no problem, makes sense to avoid breaking changes